### PR TITLE
Fixes some error on PHPStorm

### PR DESCRIPTION
**Pull request by tolbon**
Hello phppresentation team,
Some fixes reveal by static analyzer.

I have question, is it planned to migrate to PHP 7.0+ ?

### CODE REVIEW

File: src/PhpPresentation/Writer/PowerPoint2007/PptCharts.php [hunk 1]
```diff
@@ -980,7 +980,7 @@ protected function writeTypeBar(XMLWriter $objWriter, Bar $subject, $includeShee
         if ($barGrouping === Bar::GROUPING_CLUSTERED) {
             $objWriter->writeAttribute('val', '0');
         } elseif ($barGrouping === Bar::GROUPING_STACKED || $barGrouping === Bar::GROUPING_PERCENTSTACKED) {
-            $objWriter->writeAttribute('val', '100000');
```
Comment (tolbon) [inline src/PhpPresentation/Writer/PowerPoint2007/PptCharts.php]: val of overlap seem to be [-100 - 100]
```xml
<xsd:simpleType name="ST_Overlap">
    <xsd:annotation>
      <xsd:documentation>Overlap</xsd:documentation>
    </xsd:annotation>
    <xsd:restriction base="xsd:byte">
      <xsd:minInclusive value="-100" />
      <xsd:maxInclusive value="100" />
    </xsd:restriction>
  </xsd:simpleType>
```

File: src/PhpPresentation/Writer/PowerPoint2007/PptCharts.php [hunk 2]
```diff
@@ -828,6 +828,16 @@ protected function writeTypeBar(XMLWriter $objWriter, Bar $subject, $includeShee
...
+            // c:spPr
+            if ($series->getFill()->getFillType() != Fill::FILL_NONE) {
+                // c:spPr
+                $objWriter->startElement('c:spPr');
+                // Write fill
+                $this->writeFill($objWriter, $series->getFill());
+                // ## c:spPr
+                $objWriter->endElement();
+            }
+
```
Comment (tolbon) [inline src/PhpPresentation/Writer/PowerPoint2007/PptCharts.php]: Just move block seems validate by xsd

File: src/PhpPresentation/Writer/PowerPoint2007/AbstractDecoratorWriter.php [hunk 3]
```diff
@@ -51,12 +51,8 @@ protected function writeRelationship(XMLWriter $objWriter, $pId = '1', $pType =
...
      * @throws \Exception
      */
-    protected function writeBorder(XMLWriter $objWriter, $pBorder, $pElementName = 'L')
+    protected function writeBorder(XMLWriter $objWriter, Border $pBorder, $pElementName = 'L')
     {
-        if (!($pBorder instanceof Border)) {
-            return;
-        }
-
         if ($pBorder->getLineStyle() == Border::LINE_NONE && $pElementName == '') {
```
Comment (tolbon) [inline src/PhpPresentation/Writer/PowerPoint2007/AbstractDecoratorWriter.php]: Maybe I do revert this modification cause can be null and instanceof Border check this

Same for other bellow

What do you think ?

Comment (Progi1984) [reply src/PhpPresentation/Writer/PowerPoint2007/AbstractDecoratorWriter.php]: @tolbon Please revert this modification (and equivalent)

**Comment by tolbon** [comment]
I revert lot of my typing code maybe on next pull request add typing step by step
less big pr is better for reading and testing :)

**Comment by Progi1984** [comment]
@tolbon Hi, I don't understand why you change all `$this->assert` to `static::assert`. Could you explain why ?

File: src/PhpPresentation/Writer/PowerPoint2007/PptCharts.php (hunk 2 above)
Comment (Progi1984) [reply src/PhpPresentation/Writer/PowerPoint2007/PptCharts.php]: But when you generate a chart with fill, is it well displayed ?

**Comment by tolbon** [comment]
Yeah cause assert is a static function but I don’t replace everywhere do you want me to revert it ?

File: src/PhpPresentation/Writer/PowerPoint2007/PptCharts.php (hunk 2 above)
Comment (tolbon) [reply src/PhpPresentation/Writer/PowerPoint2007/PptCharts.php]: I can’t clearly say if it was that modification or change overlap to 100 not 10000.
But office365 website blank 2 bar chart for example say in issues 382.
I take latest screenshot in issue 382 and try to fix warnings/errors.

Why do you have an error ? I do modification ?

Comment (tolbon) [reply src/PhpPresentation/Writer/PowerPoint2007/PptCharts.php]: Now office 365 display all charts

**Comment by tolbon** [comment]
@Progi1984 Just a liitle Up if I need to modify some code.

**Comment by Progi1984** [comment]
@tolbon Hi, could you check the `develop` branch. I have worked on it and implemented PHPStan for static analysis.

**Comment by Progi1984** [comment]
A lot of fixes have been fixed in develop branch. Thank you for your contribution.

**Pull request closed without merging**